### PR TITLE
Add a column selector to most grids

### PR DESCRIPTION
**Pull request by Jebzou**

### CODE REVIEW

**Comment by Jebzou** [comment]
Should resolve #305

**Comment by myangelkamikaze** [comment]
The approach looks good to me.
A few notes:
1. Ok and Cancel should be flipped so that Ok is on the left side to stay consistent with the rest of the app (e.g. config)
2. Right now, if you hide all columns, you can't open the column selector anymore
3. Maybe the columns should have a "hide this column" option - selector is still needed to unhide them
4. Sorting probably shouldn't be modified in the selector - the problem is that it would be really hard to fully support the way DataGrid handles sorting (multiple sorting)
5. There could be an option to clear sorting

File: ElectronicObserver/Common/Datagrid/ColumnSelectorView.xaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+			</ui:DataGridHelper.TextColumnElementStyle>
+
+			<DataGrid.Columns>
+				<DataGridTextColumn
+					Binding="{Binding Header}"
+					Header="{Binding DataContext.ColumnSelector.Column, Source={StaticResource Proxy}}"
+					IsReadOnly="True"
+					/>
+
+				<DataGridCheckBoxColumn Binding="{Binding Visible}" Header="{Binding DataContext.ColumnSelector.Visible, Source={StaticResource Proxy}}" />
```
Comment (myangelkamikaze) [inline ElectronicObserver/Common/Datagrid/ColumnSelectorView.xaml:60]: Looks like DataGridCheckBoxColumn needs 2 clicks to actually get changed, might be better to use DataGridTemplateColumn.

https://stackoverflow.com/questions/3833536/how-to-perform-single-click-checkbox-selection-in-wpf-datagrid
```xml
<DataGridTemplateColumn Header="{Binding DataContext.ColumnSelector.Visible, Source={StaticResource Proxy}}">
	<DataGridTemplateColumn.CellTemplate>
		<DataTemplate>
			<CheckBox
				MinWidth="0"
				Margin="10 0"
				IsChecked="{Binding Visible, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}"
				/>
		</DataTemplate>
	</DataGridTemplateColumn.CellTemplate>
</DataGridTemplateColumn>
```

File: ElectronicObserver/Window/Tools/EquipmentList/EquipmentListViewModel.cs [hunk 2]
```diff
@@ -28,13 +27,12 @@ public partial class EquipmentListViewModel : WindowViewModelBase
...
-	public List<ColumnProperties> EquipmentGridColumnProperties { get; set; } = new();
-	public List<SortDescription> EquipmentGridSortDescriptions { get; set; } = new();
+	public DataGridViewModel EquipmentGridViewModel { get; set; } = new();
 	public GridLength EquipmentGridWidth { get; set; } = GridLength.Auto;
 
 	// todo: doesn't seem to work in the current implementation
-	public List<ColumnProperties> DetailGridColumnProperties { get; set; } = new();
-	public List<SortDescription> DetailGridSortDescriptions { get; set; } = new();
+	// todo: check if new implementation is working
+	public DataGridViewModel EquipmentDetaiGridViewModel { get; set; } = new();
```
Comment (myangelkamikaze) [inline ElectronicObserver/Window/Tools/EquipmentList/EquipmentListViewModel.cs:35]: Typo in EquipmentDetailGridViewModel and no, it doesn't work. Select an equipment with multiple detail items, sort by something, select a different equipment, sort data is lost.

File: ElectronicObserver/Common/Datagrid/ColumnSelectorView.xaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	</Window.Resources>
+
+	<Grid>
+
+		<Grid.RowDefinitions>
+			<RowDefinition Height="Auto" />
+			<RowDefinition />
+		</Grid.RowDefinitions>
+
+		<DataGrid
```
Comment (myangelkamikaze) [inline ElectronicObserver/Common/Datagrid/ColumnSelectorView.xaml:30]: HeadersVisibility="Column"

**Pull request merged**